client/extensions/woocommerce/app/order/order-notes/new-note.js

```diff
@@ -60,11 +59,11 @@ class CreateOrderNote extends Component {
 		return (
 			<div className="order-notes__new-note">
 				<FormFieldSet className="order-notes__new-note-content">
-					<FormLabel htmlFor="note">{ translate( 'Add a note' ) }</FormLabel>
```

Comment:
If we are removing the label, let's leave this in but wrap it in `ScreenReaderText` (I can do that, if you decide to leave off the label)